doc/application/index.rst

```diff
@@ -1446,6 +1446,156 @@ After running the preprocessor, the final device tree used in the
 build is created by running the device tree compiler, ``dtc``, on the
 preprocessor output.
 
+Building and Configuring multiple images
+****************************************
+
+When firmware update and bootloaders are involved in an embedded
+application the final firmware that is flashed onto the device will
+usually need to consist of not one executable, but multiple
+executables that chain-load, or boot, each other. This section
+describes how to build and configure such a Zephyr application.
+
+When and why to use multiple images
+===================================
+
+First off, it's necessary to have a basic understanding of what
+exactly an executable is and why one would want to have more than one
+of them. An executable goes by other names depending on context, like
+program, image, or elf file. From now on we will be using the name
+image. From these names, you probably already know what an image is,
+but in case not; An image consists of pieces of code and data that
+goes by image-unique names recorded in a single symbol table. The
+symbol table exists as metadata in an .elf or .exe file and is not
+included when converted to a hex file for device flashing. Object
+files also consist of symbols, code, and data, but object files are
+not images, because unlike object files, images have had their code
+and data placed at addresses by a linker. In the end it is the linker
+that creates images. If confused, and you want to determine if you
+have zero, one, or more images, count the number of times the linker
```

Comment:
Technically not quite true for Zephyr, right? It's actually N/2, since each image is linked twice.

Reply:
> but object files are not images, because unlike object files, images have had their code and data placed at addresses by a linker. In the end it is the linker that creates images. If confused, and you want to determine if you have zero, one, or more images, count the number of times the linker has been run

Ouch, bringing linkers into the mix makes things even more confusing... ELF files get created by a linker yet they're not images. On any x86 PC for instance, a SPI flash image contains the outputs of dozens of totally unrelated linker passes. Etc.

By mere definition (see above), any contiguous fraction of an image ("sub-image") is also an image.  Larger images are made of smaller images because the smaller parts map to parts running at different times, and/or on different parts of the chip, and/or from different code bases versioned and upgraded independently, etc. Software modularity as usual; not that much new here: https://en.wikipedia.org/wiki/Conway%27s_law

In the previously mentioned SPI flash images there's even more than one level of sub-image nesting.

I think this is why "multi-image build" is a vague name - and why there's a perceived need for this clarification attempt. What I understand this CMake effort is trying to achieve (by all means correct me) is to compile two or more (very) different configurations of the same codebase twice. So maybe "simultaneous multi-configs build" or "chain-loaded builds" would be more precise *if* being precise and formal were preferred over using informal terms users are already familiar with. To insist: I'm NOT suggesting to change the name "multi-image build". All I'm just recommending is not to bend the reality of the terms "image" and others to pretend that "multi-image builds"can be defined formally or precisely. They cannot.

So I suggest removing most of this backfiring clarification attempt, not theorizing at all and stick to the main use case / example: zephyr bootloader + zephyr OS. If other use cases apply, users will be smart enough to see them. For instance:

> Multi-image builds lets you compile the same Zephyr codebase in different ways and all in one go. Typically: 1.  bootloader. 2. Operating System. In this example, the bootloader is one "image" and the OS is the other.

Clarification: done. Nothing's formally defined yet everyone gets it in 3 lines.